### ISSUE TITLE
[bug] SystemPackageTool fails to install package lists

### ISSUE THREAD

**Issue by Jamaslab** [conan-io/conan#6960]
### Environment Details (include every applicable attribute)
  * Operating System+version: Ubuntu 18.04
  * Compiler+version: gcc 7.0
  * Conan version: 1.24.1
  * Python version: 3.6

### Steps to reproduce (Include if Applicable)
I have this conan file
 ```
def system_requirements(self):
        self.global_system_requirements=True
        installer = tools.SystemPackageTool()
        libs = "libgtk2.0-dev libavcodec-dev libavresample-dev libv4l-dev libzbar-dev libavformat-dev libswscale-dev python3-dev python3-numpy libtbb2 libtbb-dev libjpeg-dev libpng-dev libtiff-dev libdc1394-22-dev ffmpeg libvtk7-dev liblapack-dev openjdk-11-jdk"
        installer.install(libs)
```
### Logs (Executed commands with output) (Include/Attach if Applicable)
When I run conan create . I get the following output:
```
Installing (downloading, building) binaries...
dpkg-query: no packages found matching libgtk2.0-dev
dpkg-query: no packages found matching libavcodec-dev
dpkg-query: no packages found matching libavresample-dev
dpkg-query: no packages found matching libv4l-dev
dpkg-query: no packages found matching libzbar-dev
dpkg-query: no packages found matching libavformat-dev
dpkg-query: no packages found matching libswscale-dev
dpkg-query: no packages found matching libtbb2
dpkg-query: no packages found matching libtbb-dev
dpkg-query: no packages found matching libjpeg-dev
dpkg-query: no packages found matching libpng-dev
dpkg-query: no packages found matching libtiff-dev
dpkg-query: no packages found matching libdc1394-22-dev
dpkg-query: no packages found matching ffmpeg
dpkg-query: no packages found matching libvtk7-dev
dpkg-query: no packages found matching liblapack-dev
dpkg-query: no packages found matching openjdk-11-jdk
Package already installed: libgtk2.0-dev libavcodec-dev libavresample-dev libv4l-dev libzbar-dev libavformat-dev libswscale-dev python3-dev python3-numpy libtbb2 libtbb-dev libjpeg-dev libpng-dev libtiff-dev libdc1394-22-dev ffmpeg libvtk7-dev liblapack-dev openjdk-11-jdk
```
The system correctly detects that the packages are not installed but then it actually does not install them, because one of them it is.

**Comment by uilianries** [comment]
Hi @Jamaslab !

You can check if it's installed before installing:

```
installer = tools.SystemPackageTool()
libs = ["libgtk2.0-dev", "libavcodec-dev", "libavresample-dev", "libv4l-dev", "libzbar-dev", "libavformat-dev", "libswscale-dev", "python3-dev", "python3-numpy", "libtbb2", "libtbb-dev", "libjpeg-dev", "libpng-dev", "libtiff-dev", "libdc1394-22-dev", "ffmpeg", "libvtk7-dev", "liblapack-dev", "openjdk-11-jdk"]
to_be_installed = ""
for lib in libs:
    if not installer.installed(lib):
        to_be_installed += " " + lib
if to_be_installed:
    installer.install(to_be_installed)
```

Please, take a look on docs for more information:
https://docs.conan.io/en/latest/reference/conanfile/methods.html#systempackagetool

**Comment by jgsogo** [comment]
Hi, @Jamaslab 

Right now `SystemPackageTool` works using only one package at a time. We will consider this a feature here https://github.com/conan-io/conan/issues/6961

Thanks!

**Comment by Jamaslab** [comment]
Perfect, thank you very much. I would just then change this line in the documentation
https://docs.conan.io/en/latest/reference/conanfile/methods.html#method-system-requirements

install(packages, update=True, force=False): Installs the packages (could be a list or a string).

as it can't be a list.

**Comment by jgsogo** [comment]
I was wrong and you are right. The documentation says that and the implementation tries to deal with a list of packages, but it is not working as expected. This is clearly bug (not a feature, I'm closing #6961)

**Comment by memsharded** [comment]
But it is not that it cannot be a list. It can be a list of **alternatives** names for the same package. Conan will iterate them until it succeed installing one of them. 

I agree this is not clear, and the UX is bad, but it is the current implementation, and probably can't be changed without breaking.

From the docs:

> If packages is a list the first available package will be picked (short-circuit like logical or). Note: This list of packages is intended for providing alternative names for the same package, to account for small variations of the name for the same package in different distros. To install different packages, one call to install() per package is necessary.

**Comment by jgsogo** [comment]
🤕 No clear at all. So docs and implementation match (not sure), let's improve the explanation and probably modify the help string in the code as it says: `:param packages: String with all package to be installed e.g. "libusb-dev libfoobar-dev"`

Also, the implementation is able to take a list: `packages = [packages] if isinstance(packages, str) else list(packages)`, so #6961 shouldn't be needed (maybe it is not working, we need to check).

IMO we should do something here... probably it is just about improving the docs. `packages` can be a list of strings (and each string can also be a " "-separated list), I'm not sure in which coordinate/dimension I should list the alternatives.

**Comment by memsharded** [comment]
> 🤕 No clear at all. So docs and implementation match (not sure), let's improve the explanation and probably modify the help string in the code as it says: :param packages: String with all package to be installed e.g. "libusb-dev libfoobar-dev"

Yes, I really hate the way it is. I am happy if we prepare the way for a fixed one for Conan 2.0.
